src/list-ext.satyg

```diff
@@ -43,27 +37,46 @@ module List : sig
   val init : 'a list -> ('a list) option
   val reverse-append : 'a list -> 'a list -> 'a list
   val reverse-map : ('a -> 'b) -> 'a list -> 'b list
+  val all : ('a -> bool) -> 'a list -> bool
+  val any : ('a -> bool) -> 'a list -> bool
+  val zip : 'a list -> 'b list -> ('a * 'b) list
+  val unzip : ('a * 'b) list -> 'a list * 'b list
+  val find : ('a -> bool) -> 'a list -> 'a option
+  val partition : ('a -> bool) -> 'a list -> 'a list * 'a list
+  val iterate : int -> ('a -> 'a) -> 'a -> 'a list
+  val repeat : int -> 'a -> 'a list
+  val cycle- : int -> 'a list -> 'a list
+  val acons : 'a -> 'b -> ('a * 'b) list -> ('a * 'b) list
+  val max : ('a Ord.t) implicit -> 'a list -> 'a option
+  val min : ('a Ord.t) implicit -> 'a list -> 'a option
+  val fold : 'b -> ('a -> 'b -> 'b) -> 'a list -> 'b
+
+  % unstable API (may subject to change in the future)
+  val fold-left : ('a -> 'b -> 'a) -> 'a -> 'b list -> 'a
+  val fold-lefti : (int -> 'a -> 'b -> 'a) -> 'a -> 'b list -> 'a
+  val fold-right : ('a -> 'b -> 'b) -> 'b -> 'a list -> 'b
+  val fold-left-adjacent : ('a -> 'b -> 'b option -> 'b option -> 'a) -> 'a -> 'b list -> 'a
+  val map-adjacent : ('a -> 'a option -> 'a option -> 'b) -> 'a list -> 'b list
+  val mapi-adjacent : (int -> 'a -> 'a option -> 'a option -> 'b) -> 'a list -> 'b list
+  val bubblesort : ('a Ord.t) implicit -> 'a list -> 'a list
+  val insertsort : ('a Ord.t) implicit -> 'a list -> 'a list
+  val mergesort : ('a Ord.t) implicit -> 'a list -> 'a list
```

Comment:
We might add a single `sort` function and deprecate these in the future.

Reply:
Sounds good to me. We don't really care about algorithm detail. (We might want `stable-sort` like c++, but it's another discussion)

Reply:
Perhaps the sort function will call these three functions internally depending on the length of list.

Reply:
Do you mean introsort?

Reply:
Or [Timsort](https://en.wikipedia.org/wiki/Timsort) maybe?
Actually I have no concrete idea about this.